### DISCUSSION TITLE
Gitpod: When creating a workspace check node version

### DISCUSSION THREAD

**Discussion by ngillux** [freeCodeCamp/classroom#333] [General]
For some reason, there's instances where the node version is updated to the most recent one.

npm ERR! notsup Required: {"node":"^16","npm":"^8"}
npm ERR! notsup Actual:   {"npm":"9.5.1","node":"v18.16.0"}

To fix this I just ran nvm install 16 which also installs the corresponding npm version.

Additionally you may need to run npm i next after running the above if you get the below error.

sh: 1: next: not found

**Comment by GuillermoFloresV** [comment]
Hi, @ngillux thanks for opening a discussion on this. I believe we should have patched out with PR #291, was this a one-off issue? or are you able to replicate this bug consistently? If you are able to replicate it, can you post the steps you did and I can double-check this on my end? Thanks!

**Reply by ngillux** [reply]
Hi @GuillermoFloresV I'm not entirely sure that I can replicate it, but looking back on the instance this happened I think it's possible that I may have forgotten to run a step in the gitpod-instructions.sh file. I created a new workspace in Gitpod again and successfully did get the correct node version.

**Comment by lloydchang** [comment]
Hi @GuillermoFloresV and @ngillux

I chatted with @ngillux's team members via CodeDay Labs Slack about [a recurrence of this error: Required: {"node":"^16","npm":"^8"}](https://codedayorg.slack.com/archives/C05E9LWCYRZ/p1688672286582109), and we dovetailed on related topics.

**An important point that @ngillux highlighted to me [via CodeDay Labs Slack](https://codedayorg.slack.com/archives/C05E9LWCYRZ/p1688789056383539?thread_ts=1688771268.220689) is:**

> For some reason GitPod doesn’t apply the changes in .gitpod.yml and you have to explicitly run some of those commands.

[I replied to](https://codedayorg.slack.com/archives/C05E9LWCYRZ/p1688795293081569?thread_ts=1688771268.220689&cid=C05E9LWCYRZ) @ngillux as follows:

Your summary:
> For some reason GitPod doesn’t apply the changes in .gitpod.yml and you have to explicitly run some of those commands.

… makes sense to me.

It sounds like a symptom of https://github.com/gitpod-io/gitpod/issues/6720#issuecomment-970546437
> Unfortunately, Gitpod prebuilds using init: do not currently persist all changes under the home directory, just the workspace directory. See #340.
> I suggest using a customized docker image. See the docs for more information.

and https://github.com/gitpod-io/gitpod/issues/340

> Is there any way to persist the home directory? #340
> Similar to the /workspace directory?
> Most user configuration files are in the home directory, so it's quite important.
